core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -426,6 +428,38 @@ public ProxiedAddresses proxiedAddresses() {
         return proxiedAddresses;
     }
 
+    private CompletableFuture<Void> initiateConnectionShutdown(InitiateConnectionShutdown event) {
```

Comment:
How about organizing in a top-down way? i.e. public initiateConnectionShutdown methods first, and then the private one.